Requirements/cross-chain-transfer.md

```diff
@@ -38,19 +48,55 @@ Similar to the previous case, deposit transactions to the federation's multisig
 
 ![The leader comes back online](../assets/cross-chain-transfers/leader-offline-3.svg)
 
 ## 3) Building the chain B transaction deterministically
 
-The main point of building transactions deterministically is to allow for each member to independently build the exact same transaction. This way the can uniquely identify transactions coming from other members and decide to match them with their own (to collect signatures, change their statuses when finding them on chain, or when recovering from a period offline).
+There are two main reasons for building transactions deterministically.
+1. To allow for each member to independently build the exact same transaction, and therefore ensure that the federation members will never need to sign a UTXO more than once. This is for security reasons.
+2. To simplify the processes of sharing signaturs (a none determininstic approach means a leader needs to create a session for the transfer and coordinate with peers to sign it).  
+
+Assuming members are synced and have the same view of the multisig address (the collection of UTXOs) then using a predefined algorithem every member can generate the transfer transaction, sign with their key and broadcast to the other members (possibly its enought to only broadcast to the current leader).  
+It may be that in order to use a multisig UTXO the algorithem will require it to be burried under enough blocks, to avoid reorgs and ensure that all member have the same view of the UTXO list.
+
+**The algorithm to select UTXOs:**
+- The oldest UTXO first
+- If a block has more than one UTXO, then first one in the list is first
+- Outputs are sorted by destination address
+- Change address (back to multisig), if any, is last.
+- OP_RETURN of block hash of chain the transfer came from.
+
+A member that comes back online after a long perioud will need to sync both chains and also build the databse of transfer transactions, they should be able to do this simply by looking at the chains themselves.  
+A member that came online will, once it is done syncing both its chains and has the correct views of the multisig UTXO lists, check all the unprocessed transfers, start building and signing corresponding transactions, and broadcast them to its peers. 
 
 ![Building transactions](../assets/cross-chain-transfers/building-transaction.svg)
+
+**The algorithem to select the leader:**
+
+TODO:
+either reuse the current mechanism : hash block hash with public keys of members and order them alphabetically to get a leader table
+or have the leader take turn in a pre-established way based on height of block modulo federation size (somewhat similar to POA)
+
+## Considerations ## 
+
+- What is the fee going to be on the transfer trx (to maintain determinism)
+- What if a block is full of transfer transactions (too full for the other chain)
+- What if all members are offline how do we handle resync of federation members
```

Comment:
I don't see why resync would be an issue, if you assume a majority of non malicious members come back online first. Then processing the blocks in order when everybody is in sync is OK isn't it ? Maybe I am forgetting something...
If malicious members come online in a majority, then the POA can be attacked anyway I think, which means you can have transfer illegally completed (not initiated because not enough members to sign it).

Reply:
yeah I am not sure this is an issue but I put it here just so we can review it.